### DISCUSSION TITLE
java agent plugin support jdk11 problem

### DISCUSSION THREAD

**Discussion by xingcici** [apache/skywalking#8096] [Q&A] [answered]
agent version  6.x
jdk 11
maven 3.6.x

According to doc java agent support jdk 8 - jdk 17.
But when i complie agent with jdk 11, it show HttpClientWriteRequestInterceptor.java java: 程序包sun.net.www不存在 (sun.net.www does not exist).
According to jdk doc, sun.* and com.sun.* packages are not export since jdk 9+.
I found  sun.net.www.MessageHeader is still being referenced in the latest version.
![image](https://user-images.githubusercontent.com/38656144/140891637-0868e930-4549-4dae-861d-7432bb1d9b4f.png)

Has anyone encountered this problem?

**Comment by wu-sheng** [comment, marked as answer]
First of all, your codes are not matching the document. Every release has different supported versions.

6.x definitely doesn't cover all `jdk 8 - jdk 17`.

Then, the supported in readme(latest) about `8-17` is from agent core perspective. Agent plugin has its own version detection mechanism. This plugin `http-jdk-plugins` only works in `< jdk9` due to the issue you pointed out.
We are welcome you to submit a pull request to add `witness` class for old plugin, and provide a new plugin for `jdk9`.

**Reply by xingcici** [reply]
ok, thanks
